common/src/main/java/io/netty/util/ByteString.java

```diff
@@ -600,24 +602,12 @@ public boolean equals(Object obj) {
         if (!(obj instanceof ByteString)) {
             return false;
         }
-
         if (this == obj) {
             return true;
         }
-
-        ByteString that = (ByteString) obj;
-        if (length() != that.length() || hashCode() != that.hashCode()) {
-            return false;
-        }
-
-        final int end = value.length;
-        for (int i = 0, j = 0; i < end; i++, j++) {
-            if (value[i] != that.value[j]) {
-                return false;
-            }
-        }
-
-        return true;
+        AsciiString other = (AsciiString) obj;
```

Comment:
`AsciiString` -> `ByteString`.  I have unit tests that I am adding which should hit this method too...